lamindb/core/types.py

```diff
@@ -7,3 +11,23 @@
     StrField,
     TransformType,
 )
+from lamindb.core._compat import is_package_installed
+
+if TYPE_CHECKING:
+    from anndata import AnnData
+    from mudata import MuData
+    from spatialdata import SpatialData
+else:
+    from anndata import AnnData
+
+    if is_package_installed("mudata"):
```

Comment:
Have we always called all these at package import-time? At least at some point I made an effort that some things are only imported if they're needed.

Always checking for installed and then importing spatialdata seems like something that won't scale well if we want to do it with more packages in the future?